src/cyverse_groups_client/core.clj

```diff
@@ -36,7 +36,7 @@
   (grant-folder-privilege [_ user name subject privilege]
     "Grants folder privileges to a subject.")
 
-  (find-groups [_ user search] [_ user search folder]
+  (find-groups [_ user details search ] [_ user details search folder]
```

Comment:
This LGTM as-is, but I just wanted to point out an alternative I think could work, if the `details` param is intended to be optional.

This multi-method could be defined like this:
```clojure
   (find-groups [_ user search] [_ user search folder] [_ user search folder details]
     "Searches for groups by name.")
```
Then the implementations could be defined like this:
```clojure
  (find-groups [self user search]
    (find-groups self user search nil))

  (find-groups [self user search folder]
    (find-groups self user search folder nil))

  (find-groups [_ user search folder details]
    (:body (http/get (build-url base-url "groups")
...
```

Reply:
Ah, that is probably easier / clearer.  All the uses of `(partial c/find-groups ...)` in terrain made things seem complex so I just put `details` everywhere 😄 I guess I'll wait to see how the other PRs look after reviews and judge how much to change at that point.

Reply:
Now that I see how these functions are being used in `terrain`, it would probably be a lot more work to refactor `terrain` to use the functions I suggested.

So I'm fine leaving it this way for now 😵